### PR TITLE
Bug 1878664: Adding CRD and some example and documentation

### PR DESCRIPTION
**Pull request by joselsegura**

### CODE REVIEW

File: docs/custom-resource-sample/report_overview1.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: "external-pipeline.redhat.com/v1"
+kind: ReportOverview
+metadata:
+  name: my-new-report-overview-object
```
Comment (bond95) [inline docs/custom-resource-sample/report_overview1.yaml:4]: nit-pick: I'm not sure about this name maybe better `report-overview-object`

**Review by tisnik — APPROVED** [review]
LGTM

File: docs/custom-resource-sample/report_overview1.yaml (hunk 1 above)
Comment (joselsegura) [reply docs/custom-resource-sample/report_overview1.yaml:4]: I'll update it, but it is just an example :-D

File: manifests/08-report_overview_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: report-overviews.external-pipeline.redhat.com
+spec:
+  group: external-pipeline.redhat.com
+  versions:
+    - name: v1
```
Comment (iNecas) [inline manifests/08-report_overview_crd.yaml:8]: I would be more conservative with marking this `v1alpha1` at this point.

**Review by natiiix — COMMENTED** [review]
I don't mind anything personally, but the version should probably be alpha/beta as Ivan suggested.

**Comment by joselsegura** [comment]
Agree on @iNecas and @natiiix comments. I'll update it ASAP

File: manifests/08-report_overview_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: report-overviews.external-pipeline.redhat.com
+spec:
+  group: external-pipeline.redhat.com
```
Comment (martinkunc) [inline manifests/08-report_overview_crd.yaml:6]: Woudln't it make better sense to call the group for example "insights.redhat.com" ? This looks a bit dependent on one particular place. What do you think ?

File: manifests/08-report_overview_crd.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                moderate:
+                  type: integer
+                important:
+                  type: integer
+                critical:
+                  type: integer
+  scope: Cluster
+  names:
+    plural: report-overviews
+    singular: report-overview
```
Comment (martinkunc) [inline manifests/08-report_overview_crd.yaml:29]: It looks like most of the multiword names doesn't use dash, would "reportoverview" make sense ? List crds for example here: http://pastebin.test.redhat.com/901861

File: manifests/08-report_overview_crd.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                low:
+                  type: integer
```
Comment (martinkunc) [inline manifests/08-report_overview_crd.yaml:19]: Could we add Description: What does Low property mean ..

**Comment by joselsegura** [comment]
@smarterclayton could you please take a look on this PR? It adds a new CRD to the clusters to be used by Insights Operator. 

Thank you in advance

File: docs/custom-resource-sample/report_overview1.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+apiVersion: "insights.redhat.com/v1alpha"
+kind: ReportOverview
```
Comment (tremes) [inline docs/custom-resource-sample/report_overview1.yaml:2]: Isn't this too general? Wouldn't be better e.g `InsightsReportOverview`? Just asking.... :-)

**Pull request closed without merging**